### ISSUE TITLE
Alignment with Dokieli's HTML patterns

### ISSUE THREAD

**Issue by rubensworks** [rubensworks/ScholarMarkdown#2]
We should look into making the resulting HTML of ScholarMarkdown match with that of the Dokieli editor as much as possible. They are already slightly similar, but there seem to be some difference regarding main/content.

That would allow us to share CSS files as-is.

**Comment by rubensworks** [comment]
The current output of an article in ScholarMarkdown looks something like this:
```html
<header>
  <h1 id="the-fundamentals-of-semantic-versioning">The Fundamentals of Semantic Versioning</h1>

  <ul id="authors">
    <li><a href="http://www.rubensworks.net/" typeof="http://xmlns.com/foaf/0.1/Person" resource="http://www.rubensworks.net/#me">Ruben Taelman</a></li>
  </ul>

  <ul id="affiliations">
    <li id="idlab">IDLab,
          Department of Electronics and Information Systems,
          Ghent University – imec</li>
  </ul>

  <section id="abstract">
    <h2>Abstract</h2>
  </section>

</header>
<main>
  <section id="introduction">
    <h2>Introduction</h2>

    <p>This is a great article on the Semantic Web <a href="#ref-1" class="reference">[1]</a>, written in Markdown.</p>

  </section>

  <section id="conclusions">
    <h2>Conclusions</h2>

  </section>

</main>

<footer>
    <section id="references">
    <h2>References</h2>
    <dl class="references">
      <dt id="ref-1">[1]</dt>
      <dd resource="#semanticweb" typeof="schema:Article">Berners-Lee, T., Hendler, J., Lassila, O., others: The semantic web.  Scientific American. 284, 28–37 (2001).</dd>
      </dl>
      </section>
</footer>
```

While the general output of Dokieli is:
```html
    <main>
      <article>
                <h1 property="schema:name">Decentralised Authoring, Annotations and Notifications for a Read-Write Web with dokieli</h1>

                <div id="authors">
                    <dl id="author-name">
                        <dt>Authors</dt>
                        <dd id="Sarven-Capadisli" inlist="" rel="bibo:authorList" resource="http://csarven.ca/#i"><span about="" rel="schema:creator schema:publisher schema:author"><a about="http://csarven.ca/#i" href="http://csarven.ca/" property="schema:name" rel="schema:url" typeof="schema:Person">Sarven Capadisli</a></span><sup><a about="http://csarven.ca/#i" href="#University-of-Bonn" rel="schema:memberOf" resource="http://dbpedia.org/resource/University_of_Bonn">1</a></sup><sup><a href="#email-Sarven-Capadisli">✊</a></sup></dd>
                    </dl>

                    <ul id="author-org">
                        <li id="University-of-Bonn"><sup>1</sup>Enterprise Information Systems Department, <a about="http://dbpedia.org/resource/University_of_Bonn" href="http://uni-bonn.de/" property="schema:name" rel="schema:url" typeof="schema:Organization">University of Bonn</a>,  Bonn, DE</li>
                    </ul>

                    <ul id="author-email">
                        <li id="email-Sarven-Capadisli"><sup>✊</sup><a about="http://csarven.ca/#i" href="mailto:info@csarven.ca" rel="schema:email">info@csarven.ca</a></li>
                    </ul>
                </div>

                <div id="content">
                    <section id="abstract">
                        <h2>Abstract</h2>
                    </section>

                    <section id="introduction" inlist="" rel="schema:hasPart" resource="#introduction">
                        <h2 property="schema:name">Introduction</h2>
                    </section>

                    <section id="references">
                        <h2>References</h2>
                        <div>
                            <ol>
                                <li>Otto, B., Auer, S., et al: <cite>Industrial Data Space</cite>, Technical Report, 2016, <a href="https://www.researchgate.net/publication/299536747_Industrial_Data_Space_Digital_Souvereignity_Over_Data" rel="schema:citation">https://www.researchgate.net/publication/299536747_Industrial_Data_Space_Digital_Souvereignity_Over_Data</a></li>
                            </ol>
                        </div>
                    </section>
                </div>

      </article>
    </main>

    <footer>
    </footer>
```

The main differences are:
* ScholarMarkdown uses the `<header />`, `<main />` and `<footer />`, while Dokieli uses sections inside the `<article />`.
* Authors are formatted differently: ol (ScholarMarkdown) vs dl (Dokieli).
* References are formatted differently: dl (ScholarMarkdown) vs ol (Dokieli).

@csarven, does my interpretation of Dokieli look correct to you?

**Comment by RubenVerborgh** [comment]
I like authors as a list; I think this is what they really are. The `dl` in Dokieli seems overkill.

I do like `article`, it's just crazy that HTML5 thinks you cannot have `header` in there (or something, I remember there was a blocker like that).

Regarding references, no strong opinion, except that some citation styles use different keys beside numbers; I wanted to explicitize the correspondence between the in-text references and those in the references section. But such lookup-table-based references are a thing of paper anyway; they should just become hyperlinks in the future.

**Comment by csarven** [comment]
I think the header/footer is a non-issue. It can include or exclude them. There is nothing that I'm aware of in dokieli (CSS or JS) that particularly depends on them. There are some articles that use a header eg. http://semstats.org/2018/call-for-contributions , https://linkedresearch.org/events/the-web-conf-2018/ and likewise for footer. If there are any features that do not work due to things like footer/header missing, I would consider that as a bug in dokieli, or something that should be ironed out.

I find `main > article` to be a good indicator as the name applies... "main article" of the document - there may be many... and for example, when annotations are injected into the document, they actually contain the `article` node (inside an aside), .. or say interactions/comments listed at the bottom of the article. In any case, not having `main > article` is not a problem - dokieli has a fallback eg. main > article, main, body for certain operations.

The dl for authors and other document level stuff was intended to have a heading for the data so that there is some indication as to what they are - especially when CSS/JS is not involved. ol is fine too. There is nothing on the JS end that cares about that. Some of the CSS (eg acm/lncs) however does IIRC.

For references.. I don't have a strong preference either. Used li in dokieli to keep it simple. They have an identifier which correspond to the internal anchor elsewhere in the document. I like the dt/dd approach as well in that of having an explicit key, but that can also be in the li... in any case, dokieli's position so far with the print-like references was to just have the basics in place, and simple styling.. the interesting bits are left for RDFa to handle. I don't want to expand on that point too much here because there many ways to do the citations (and properly in the body of the article that's specific to the ideas) ... and so what's visible in the "References" section is usually the basic linking to the cited object in question - not so much about redescribing the object's data.

**Comment by rubensworks** [comment]
> I find `main > article` to be a good indicator as the name applies.

True, that makes sense. We should be able to add an `<article />` inside `<main />` without breaking anything. The only downside is that the authors and references won't be contained in the article tag. It's either that, or removing header/footer.

> I like the dt/dd approach as well in that of having an explicit key, but that can also be in the li...

I might make sense to also just use `<li />` here, for consistency with the authors list. The only difference with Dokieli would then be the inclusion of explicit print-like references.  @RubenVerborgh we'll probably need some kind of serializer option in BibMarkdown for this?

@csarven Is there a possibility to use other citation styles in Dokieli? If not, would it make sense to use such explicit print-like references in Dokieli as well? Then you wouldn't be dependent on the numerical identifiers as generated by `<li />`.

**Comment by csarven** [comment]
Oh I see. The header/footer that I've used was for the document as a whole, not for main > article. body > header/footer comes in handy for cross document (like site wide stuff eg. nav, address). For information that's really specific to the "article" at hand (eg. like the authors), I tend to put that in main > article. The differences may not be a big deal here for your usage but just mentioning the decision behind dokieli (and how it is used in different documents). I wanted the HTML patterns to be reusable, and get as much CSS stuffreuse out of it as well.

Again, if you prefer, you can leave the header/footer as you have (outside of main), and even not bother to include article. I don't think anything would be particularly an issue there. main just indicates, you know, "core info about this document right here" -- that's why I figured to keep the author/references etc.. closer to the main content (the article).

Using ol (or ul) for authors is fine but keep in mind that some of hte stylesheets are based off dl. I guess one thing that can be done is update the stylesheets to use the RDFa patterns. This is all so that, the default stylesheet for ScholarMarkdown can do its own thing, and when the config is in place to say include additional stylesheets and to be able to swap it will all "just work".

The citation styles as in their counter value? Yes, that'd be dependent on each stylesheet. The dl approach is good in that, the dt forces the key in the content, where as in the li approach it is not in the HTML but handled by CSS instead. I went back and forth on this in the past.. just didn't care enough to strongly commit to one other than taking the simple approach. Don't forget that having the dt as key is a particular kind of citation / the default. So, switching to say (alpha)numeric would require change in content. The li approach is nice in that, only the CSS needs to be change, but of course with the limitation in that the intended key is not used - just an autocounter.

**Comment by RubenVerborgh** [comment]
> I might make sense to also just use <li /> here, for consistency with the authors list.

Consistency is not the right argument here: authors are a list. References are a list or a key/value table, depending on how you look at it. (Authors are not key/value.)

**Comment by csarven** [comment]
You you can use ol/ul for the authors if you like, but there is nothing wrong with dl. See example: https://www.w3.org/TR/html/grouping-content.html#example-e79dae5b or even the editors at https://www.w3.org/TR/html/ using dl.

**Comment by RubenVerborgh** [comment]
Yeah, but that's with multiple dt elements. As it currently stands, I don't
see the added value.

**Comment by csarven** [comment]
dl is flexible as to how the key/values are grouped. The point was dl not being wrong re "authors are not key/value". dt provides the label to indicate what the "list" is. IMO, that is the added value. Okay to do something similar with hX + ol as long as there is some hint - a paragraph saying "here are the authors" would just as well -  as to what the list is about. Certainly there is no requirement to have a text before the list either.. just HTML being flexible about how people want to author and make it comprehensible for their readers. IMO, can't go super wrong whether there is a label of some sort up front on ol/ul, or dl, or even just having a paragraph with commas between author names.

**Comment by RubenVerborgh** [comment]
Not wrong, just not useful. A definition list with one definition is as
useful as a bulleted list with one bullet.

**Comment by csarven** [comment]
Clearly the dl is useful for "zero or more term-description groups" as per HTML. Regardless of how many dds a dl has, its dt(s) provide some human-readable text for the corresponding items.

To compare that to ul/ol, it would need some preceding node, eg. hX, p, .. doesn't matter. If that's not required or wanted from the ul/ol approach, it is not particularly useful to compare in my opinion.

dl provides an opportunity for the whole - conceptually related - block to be carried around, manipulated, styled eg. visually grouped information, copy/paste of the rendered text, easy to identify the single block (`id=""`) to remove/update or to even provide sufficient flexibility on how it could be styled. The decision being based on content is certainly the driving factor but it is not strictly the case - it is a holistic view on what markup should be used. So, having some plans on how/where else similar types of content will be used helps.

Adding/removing items to all kinds of lists are common operations. Sometimes the prose hints at that possibility. No need to force a single item into a paragraph for example - not that there is anything wrong with doing that. Neither do we have to come up with different HTML patterns and its template management just to handle a single or multiple authors - a list will be just fine for both cases most of the time because the alternative would be overkill and not necessarily most reusable considering everything.

It depends a lot on the author's comfort in what they want. IIRC, different ways to markup - not necessarily far fetched - is widely accepted in the HTML spec circle.

**Comment by RubenVerborgh** [comment]
Thanks for taking the effort to explain—no strong opinion about it anymore now.
